server/game/core/card/Card.ts

```diff
@@ -205,77 +210,42 @@ export class Card extends OngoingEffectSource {
         return [args[0] as Player, args[1]];
     }
 
-    /**
-     * Create card abilities by calling subsequent methods with appropriate properties
-     * @param {Object} ability - AbilityHelper object containing limits, costs, effects, and game actions
-     */
-    // eslint-disable-next-line @typescript-eslint/no-empty-function
-    protected setupCardAbilities(ability) {
-    }
-
-    /**
-     * Works through the list of queued ability initializers and activates any for the corresponding type.
-     * We need to initialize this way because subclass ability initializers might be called before their
-     * constructors have executed, so we have to delay execution of their initializers until they're ready.
-     */
-    protected activateAbilityInitializersForTypes(abilityTypes: AbilityType | AbilityType[]) {
-        const abilityTypesAra = Helpers.asArray(abilityTypes);
-
-        const skippedInitializers: IAbilityInitializer[] = [];
-        for (const abilityInitializer of this.abilityInitializers) {
-            if (abilityTypesAra.includes(abilityInitializer.abilityType)) {
-                abilityInitializer.initialize();
-            } else {
-                skippedInitializers.push(abilityInitializer);
-            }
-        }
-        this.abilityInitializers = skippedInitializers;
-    }
-
-    protected addActionAbility(properties: IActionAbilityProps<this>) {
-        this.abilityInitializers.push({
-            abilityType: AbilityType.Action,
-            initialize: () => this._actionAbilities.push(this.createActionAbility(properties))
-        });
-    }
-
     public createActionAbility(properties: IActionAbilityProps): ActionAbility {
         properties.cardName = this.title;
         return new ActionAbility(this.game, this, properties);
     }
 
 
     // ******************************************* CARD TYPE HELPERS *******************************************
-    // TODO: convert these to use ts type narrowing for simpler conversions to derived types (see https://www.typescriptlang.org/docs/handbook/2/classes.html#this-based-type-guards)
-    public isEvent(): boolean {
+    public isEvent(): this is EventCard {
```

Comment:
I love/hate the syntax here lol

Reply:
Yeah it's a little clunky but it solves a real problem that's kind of unique to TS